akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -522,8 +522,9 @@ final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out) extends GraphSta
           pull(in)
         } catch {
           case NonFatal(ex) ⇒ decider(ex) match {
-            case Supervision.Stop ⇒ failStage(ex)
-            case _ ⇒
+            case Supervision.Stop   ⇒ failStage(ex)
```

Comment:
That last one would be one extra allocation for every element passing through though. Better use try-catch in this case.